### DISCUSSION TITLE
Shape only systematics

### DISCUSSION THREAD

**Discussion by Tomoya-Iizawa** [scikit-hep/cabinetry#445] [Q&A] [answered]
Hi,

I would like to implement systematics only with shape, since normalization uncertainty is included as overall uncertainty.
Is there any option for this?
In the document, I can find only Normalization and NormPlusShape.
Is it necessary to normalize variation distribution to nominal distribution by hand?

Best,
Tomoya

**Comment by alexander-held** [comment, marked as answer]
Hi Tomoya, there is currently no option for this. I think this should be in scope for `cabinetry` though and not too difficult to implement. I added #447 to track this.

The easiest way at the moment would probably be to have a custom post-processing script that removes `normsys` modifiers from the workspace for all the cases where you would like to use such systematics. Below is an example which uses the example workspace created by [`example.py`](https://github.com/scikit-hep/cabinetry/blob/d5e22413a963d38a9c945ad83ce7e8dc1c6af3e7/example.py):

```python
import cabinetry

ws = cabinetry.workspace.load("workspaces/example_workspace.json")

MOD_NAMES_TO_PRUNE = ["Modeling"]  # modifiers name to prune
MOD_TYPE_TO_PRUNE = ["normsys"]  # modifier types to prune

# the ws object is just a dictionary, iterate through that and drop pieces as needed
# only prune modifiers matching all requirements (name and type)
for channel in ws["channels"]:
    for sample in channel["samples"]:
        sample["modifiers"] = [
            mod
            for mod in sample["modifiers"]
            if not (
                mod["name"] in MOD_NAMES_TO_PRUNE and mod["type"] in MOD_TYPE_TO_PRUNE
            )
        ]

model, data = cabinetry.model_utils.model_and_data(ws)
```

You can check e.g. with `cabinetry.visualize.modifier_grid(model)` before and after this pruning that the edits work as intended.

In `pyhf` there is a function to perform pruning, however I think it does not work too well for this specific case: you want to prune only if multiple requirements are fulfilled (modifier type is `normsys` and name matches that of the systematic you want to prune). At the moment, `pyhf` prunes with a logical or (see https://github.com/scikit-hep/pyhf/issues/2374).

**Reply by Tomoya-Iizawa** [reply]
Hi,
Thank you for your answer and sorry for the late reply.
I will refer to that, and for the time being, I made a custom script to remove the effect of normalization from variation distribution.
